pkg/core/resources/apis/mesh/dataplane_overview.go

```diff
@@ -125,22 +128,46 @@ func (d *DataplaneOverviewResourceList) RetainMatchingTags(tags map[string]strin
 	d.Items = result
 }
 
+func modeToFilter(mode string) DpFilter {
+	isnil := func(a interface{}) bool {
+		return a == nil || reflect.ValueOf(a).IsNil()
+	}
+	switch mode {
+	case "true":
+		return func(a interface{}) bool {
+			return !isnil(a)
+		}
+	case "false":
+		return func(a interface{}) bool {
+			return isnil(a)
+		}
+	default:
+		return func(a interface{}) bool {
```

Comment:
If a query has `?gateway=some_garbage_value` does it mean it returns gateways? Can we somehow validate this?